src/test_application/activescenes/common_scene.h

```diff
@@ -0,0 +1,140 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2022 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#pragma once
+
+#include <osp/Active/basic.h>
+#include <osp/Active/drawing.h>
+
+//#include <entt/core/any.hpp>
+
+#include <array>
+
+namespace testapp
+{
+
+/**
+ * @brief An array of entt::any intended for storing unique types
+ */
+struct MultiAny
+{
+    using Array_t = std::array<entt::any, 8>;
+
+    Array_t m_data;
+
+    template <typename T, typename ... ARGS_T>
+    T& emplace(ARGS_T&& ... args);
+
+    template <typename T>
+    T& get();
+};
+
+template <typename T, typename ... ARGS_T>
+T& MultiAny::emplace(ARGS_T&& ... args)
+{
+    Array_t::iterator found = std::find_if(std::begin(m_data), std::end(m_data),
+                              [] (entt::any const &any)
+    {
+        return ! bool(any);
+    });
+
+    assert (found != std::end(m_data));
+
+    found->emplace<T>(std::forward<ARGS_T>(args) ...);
+
+    return entt::any_cast<T&>(*found);
```

Comment:
I feel like you would want to mimic the api of std::set or std::map, with this function returning a std::pair<iterator, bool>

Reply:
Hmm, it should also detect if the type already exists, and return false there.